design/pub-sub.md

```diff
@@ -0,0 +1,42 @@
+---
+title: pub-sub
+target-version: 0.1.0
+---
+
+# PubSub
+
+## Summary
+Yorkie needs to share events happening in documents between different clients. For this, we implement this feature using gRPC server-side stream and the PubSub pattern.
+
+### Goals
+We should be able to share events with other clients who are subscribing to Documents.
+
+## Proposal Details
+
+### How does it work?
+Yorkie implements WatchDocuments API using [gRPC server-side stream](https://grpc.io/docs/languages/go/basics/#server-side-streaming-rpc) to deliver the events that have occurred to other clients.
```

Comment:
Yorkie implements WatchDocuments API using [gRPC server-side streaming](https://grpc.io/docs/languages/go/basics/#server-side-streaming-rpc) to deliver the events that have occurred to other clients.